crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -1701,25 +1711,38 @@ impl<Codec: RawEntryCodec> StateMachine<Codec> {
                 ServiceInvocationResponseSink::PartitionProcessor {
                     entry_index,
                     caller,
-                } => self.handle_outgoing_message(ctx, OutboxMessage::ServiceResponse(InvocationResponse {
-                    id: caller,
-                    entry_index,
-                    result: result.clone(),
-                })).await?,
-                ServiceInvocationResponseSink::Ingress { node_id, request_id } => {
-                    Self::send_ingress_response(ctx, IngressResponseEnvelope{ target_node: node_id, inner: ingress::InvocationResponse {
-                        request_id,
-                        invocation_id,
-                        response: match result.clone() {
-                            ResponseResult::Success(res) => {
-                                IngressResponseResult::Success(invocation_target.expect("For success responses, there must be an invocation target!").clone(), res)
-                            }
-                            ResponseResult::Failure(err) => {
-                                IngressResponseResult::Failure(err)
-                            }
-                        },
-                    } })
+                } => {
+                    self.handle_outgoing_message(
+                        ctx,
+                        OutboxMessage::ServiceResponse(InvocationResponse {
+                            id: caller,
+                            entry_index,
+                            result: result.clone(),
+                        }),
+                    )
+                    .await?
                 }
+                ServiceInvocationResponseSink::Ingress {
+                    node_id,
+                    request_id,
+                } => Self::send_ingress_response(
```

Comment:
Are we still needing the `Ingress` variant? Given that a rpc request is ephemeral, we probably don't need to persist that an ingress is waiting for a response.